rero_ebooks/api.py

```diff
@@ -49,31 +67,86 @@ class Ebook(Record):
     uri_key = 'electronic_location_and_access'
 
     @classmethod
-    def create_or_update(
-        cls,
-        data,
-        id_=None,
-        dbcommit=False,
-        reindex=False,
-        vendor=None,
-        **kwargs
-    ):
+    def _merge_uri(cls, new_record, old_record):
+        """Merge new and old records."""
+        field = cls.uri_key
+        new_e_res = new_record.get(field)
+        # change all tuples to lists
+        # the dojson produces tuples and we have lists in the record
+        for e_res in new_e_res:
+            for key, value in e_res.items():
+                if isinstance(value, tuple):
+                    e_res[key] = list(value)
+
+        old_e_res = old_record.get(field)
+        for e_res in old_e_res:
+            # check if already exists!
+            if e_res not in new_e_res:
+                new_e_res.append(copy.deepcopy(e_res))
+                idx = new_record['__order__'].index(field)
+                new_record['__order__'].insert(idx, field)
+        return new_record
+
+    @classmethod
+    def create_or_update(cls, data, id_=None, dbcommit=False, reindex=False,
+                         vendor=None, **kwargs):
```

Comment:
please describe the input parameters in the docstring. Please check also other methods.